enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,388 @@
+ovnkube-node-dpu-host daemonset pods onto DPU hosts.
+
+A new configmap `ovn-management-ports` will be created by SR-IOV Network
+Operator, this configmap is mounted to CNO ovnkube-node-dpu-host daemonset
+to pass node ovn management port information.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ovn-management-ports
+  namespace: openshift-ovn-kubernetes
+data:
+  dpu-host-worker-0: '{"ovnkube-node-mgmt-port":"eth0v0"}'
+  dpu-host-worker-1: '{"ovnkube-node-mgmt-port":"eth0v0"}'
+```
+
+There are three supported modes with NVIDIA DPU BlueField2:
+- **Separated host (default)**: The embedded ARM system and the function
+exposed to the host are both symmetric. Each one of those functions has its
+own MAC address and is able to send and receive packets.
+- **Embedded CPU**: The embedded ARM system controls the NIC resources and
+data path. A network function is still exposed to the host, but it has limited
+privileges.
+- **Restricted**: Also referred to as isolated mode, for security and isolation
+purposes, it is possible to restrict the host from performing operations that
+can compromise the DPU.
+
+BlueField2 needs to be configured to **Embedded CPU** or **Restricted** mode
+when used for OVN hardware offload. SR-IOV network config daemon, a daemonset
+managed by SR-IOV Network Operator, will be updated to use mstflint version >=
+`4.15.0-1.el8` which supports BlueField2 mode configuration.
+
+SR-IOV Network Operator identifies the NIC using its device ID, the NVIDIA
+BlueField2 device ID will be recoganized in SR-IOV Network Operator.
+
+SR-IOV Network Operator will move to use systemd service for configuring
+devices vs configuring devices in SR-IOV config daemonset pods. This ensures
+that SR-IOV device with VF index 0 (which is used as ovn management port)
+becomes available before ovnkube-node-smart-nic pod starts.
+
+SR-IOV Network Operator will apply udev rule to DPU host VF0 to ensure the VF
+interface name is consistant across rebooting and kernel upgrade.
+
+### Risks and Mitigations
+
+#### Setting up initial DPU connectivity
+
+It is required to establish an initial transparent network connection between
+DPU host and external network through DPU ARM cores in order to deploy DPU
+host as an OpenShift worker node. This will be addressed by deploying DPU
+as a worker node in a separate infrastructure cluster and applying network
+configurations via MachineConfig in that cluster.
+
+#### BlueField2 OOB interface driver not in RHEL
+
+Accessing to BlueField2 ARM cores via Out-Of-Band (OOB) interface requires
+additional [mlxbf_gige driver](https://bugzilla.redhat.com/show_bug.cgi?id=1803489) which is not available in RHEL-8.4.
+This is currently done through ssh or console from DPU host using rshim
+[userspace driver](https://bugzilla.redhat.com/show_bug.cgi?id=1744737).
+Rshim userspace driver needs to be made availabe in RHCOS image or downloadable
+in Universal Base Image (UBI).
```

Comment:
We have no plans to use rshim even in dev preview (right?), so this doesn't need to be mentioned.

Reply:
Right, removed rshim.